src/server/src/Server.cpp

```diff
@@ -201,6 +203,40 @@ std::string Server::track_position(const double x, const double y, const double
 }
 
 
+std::string Server::set_position_track_param(const double amax, const double vmax, const double jerk)
+{
+    State state = get_state();
+
+    if (state != State::Idle)
+        set_state(State::Idle);
+        
+    set_state(State::SetPosTrackParam);
+
+    amax_ = amax;
+    vmax_ = vmax;
+    jerk_ = jerk;
+
+    return "OK";
+}
+
+
+std::string Server::set_position_move_param(const double amax, const double vmax, const double jerk)
+{
+    State state = get_state();
+
+    if (state != State::Idle)
```

Comment:
What is the rationale of this and following lines?

If the state is different from Idle it is set to Idle. However, later you change it to SetPosTrackParam.

Reply:
right, that was what I understood you want me to insert. Took it out.